### ISSUE TITLE
Docker serve

### ISSUE THREAD

**Issue by zepouet** [squidfunk/mkdocs-material#2168]
Hello,

Simply but following the documentation the two command fails.

```
docker run --rm -it -v ${PWD}:/docs squidfunk/mkdocs-material new .
docker run --rm -it -p 8000:8000 -v ${PWD}:/docs squidfunk/mkdocs-material
```

The error is 

```
INFO    -  Building documentation...
Traceback (most recent call last):
  File "/usr/local/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/mkdocs/__main__.py", line 133, in serve_command
    serve.serve(
  File "/usr/local/lib/python3.8/site-packages/mkdocs/commands/serve.py", line 141, in serve
    config = builder()
  File "/usr/local/lib/python3.8/site-packages/mkdocs/commands/serve.py", line 122, in builder
    config = load_config(
  File "/usr/local/lib/python3.8/site-packages/mkdocs/config/base.py", line 197, in load_config
    errors, warnings = cfg.validate()
  File "/usr/local/lib/python3.8/site-packages/mkdocs/config/base.py", line 115, in validate
    post_failed, post_warnings = self._post_validate()
  File "/usr/local/lib/python3.8/site-packages/mkdocs/config/base.py", line 95, in _post_validate
    config_option.post_validation(self, key_name=key)
  File "/usr/local/lib/python3.8/site-packages/mkdocs/config/config_options.py", line 469, in post_validation
    config[key_name] = theme.Theme(**theme_config)
  File "/usr/local/lib/python3.8/site-packages/mkdocs/theme.py", line 45, in __init__
    self._load_theme_config(name)
  File "/usr/local/lib/python3.8/site-packages/mkdocs/theme.py", line 75, in _load_theme_config
    theme_dir = utils.get_theme_dir(name)
  File "/usr/local/lib/python3.8/site-packages/mkdocs/utils/__init__.py", line 297, in get_theme_dir
    return os.path.dirname(os.path.abspath(theme.load().__file__))
  File "/usr/local/lib/python3.8/posixpath.py", line 374, in abspath
    path = os.fspath(path)
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

Last version of Docker on MacOSX BigSur.

Thanks a lot.
Best regards,

**Comment by squidfunk** [comment]
Does it happen on Catalina? I'm afraid I can't debug BigSur, as I don’t have it.

**Comment by zepouet** [comment]
Hello Martin,

Thanks to answer the first of january 👍 
I reproduce on Catalina. It is more a problem about path into image to my point of view than Docker version or OS.

I changed the entrypoint and I have always the same error.

I will push later you an example with K8S to generate documentation from Gitlab-CI.

Thanks.

```
docker run --entrypoint sh --rm -it -p 8000:8000 -v ${PWD}:/docs squidfunk/mkdocs-material

/docs # mkdocs serve
INFO    -  Building documentation...
Traceback (most recent call last):
  File "/usr/local/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/mkdocs/__main__.py", line 133, in serve_command
    serve.serve(
  File "/usr/local/lib/python3.8/site-packages/mkdocs/commands/serve.py", line 141, in serve
    config = builder()
  File "/usr/local/lib/python3.8/site-packages/mkdocs/commands/serve.py", line 122, in builder
    config = load_config(
  File "/usr/local/lib/python3.8/site-packages/mkdocs/config/base.py", line 197, in load_config
    errors, warnings = cfg.validate()
  File "/usr/local/lib/python3.8/site-packages/mkdocs/config/base.py", line 115, in validate
    post_failed, post_warnings = self._post_validate()
  File "/usr/local/lib/python3.8/site-packages/mkdocs/config/base.py", line 95, in _post_validate
    config_option.post_validation(self, key_name=key)
  File "/usr/local/lib/python3.8/site-packages/mkdocs/config/config_options.py", line 469, in post_validation
    config[key_name] = theme.Theme(**theme_config)
  File "/usr/local/lib/python3.8/site-packages/mkdocs/theme.py", line 45, in __init__
    self._load_theme_config(name)
  File "/usr/local/lib/python3.8/site-packages/mkdocs/theme.py", line 75, in _load_theme_config
    theme_dir = utils.get_theme_dir(name)
  File "/usr/local/lib/python3.8/site-packages/mkdocs/utils/__init__.py", line 297, in get_theme_dir
    return os.path.dirname(os.path.abspath(theme.load().__file__))
  File "/usr/local/lib/python3.8/posixpath.py", line 374, in abspath
    path = os.fspath(path)
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

**Comment by zepouet** [comment]
Sorry I am stupid...

I didn't add

```
theme:
  name: material
```

Sorry to waste your time and thanks for your work. Thanks again

**Comment by squidfunk** [comment]
I investigated this error - it stems from the fact that we remove the default images from the Docker container:

https://github.com/squidfunk/mkdocs-material/blob/64e9a496f604ebfa1fb2ce740b0379993ca8fd96/Dockerfile#L57

Why? Because it saves us 10 MB of the packed size of the Docker image. Since this is the official Material for MkDocs Docker image, it shouldn't be used with the default themes, only with MkDocs. The reported error is the result of MkDocs trying to use the default `mkdocs` theme, which is not there anymore. Setting `material` as a theme fixes it.

64e9a496 tricks MkDocs in thinking that both default images are still available, but in fact they're only symlinks. This fixes the issue which I just experienced when leading a client through the Docker setup.
